tests/main/snap-disconnect/task.yaml

```diff
@@ -43,3 +55,46 @@ execute: |
 
     echo "Disconnecting again using abbreviated form is handled"
     snap disconnect home-consumer:home | MATCH "No connections to disconnect"
+
+    inspect_connection() {
+      CONN="$1"
+      # shellcheck disable=SC2002
+      cat /var/lib/snapd/state.json | jq -r '.data["conns"] | has("'"$CONN"'")'
+    }
+
+    # these checks rely on automatic connection of home on non-core systems
+    if ! is_core_system; then
+      echo "Checking that --forget forgets connection when auto-connected"
+      snap remove --purge "$SNAP_FILE"
+      snap install --dangerous "$SNAP_FILE"
+      snap connections home-consumer | MATCH "home-consumer:home *:home"
+      snap disconnect --forget home-consumer:home :home
+      snap interfaces | grep -Pzqe "$DISCONNECTED_PATTERN"
+      inspect_connection "home-consumer:home core:home"| MATCH "false"
+
+      echo "Checking that --forget forgets the connection when already disconnected"
+      snap remove --purge "$SNAP_FILE"
+      snap install --dangerous "$SNAP_FILE"
+      # home got auto-connected
+      snap connections home-consumer | MATCH "home-consumer:home *:home"
+      snap disconnect home-consumer:home :home
+      # still remembered as automatic connection is now undesired
+      inspect_connection "home-consumer:home core:home" | MATCH "true"
+      snap interfaces | grep -Pzqe "$DISCONNECTED_PATTERN"
+      snap disconnect --forget home-consumer:home :home
+      inspect_connection "home-consumer:home core:home" | MATCH "false"
+    fi
+
+    echo "Checking that a connection for missing plug can be forgotten"
+    snap install --dangerous test-snap-producer_1_all.snap
```

Comment:
could we use the `install_local` function here instead?

Reply:
Done.